src/workerd/api/streams/internal.c++

```diff
@@ -1613,6 +1613,14 @@ bool WritableStreamInternalController::Pipe::checkSignal(jsg::Lock& js) {
   KJ_IF_SOME(signal, maybeSignal) {
     if ((signal)->getAborted()) {
       auto reason = signal->getReason(js);
+
+      // abort process might call parent.drain which will delete this,
+      // move/copy everything we need after into temps.
+      auto& parent = this->parent;
+      auto& source = this->source;
+      auto preventCancel = this->preventCancel;
+      auto promise = kj::mv(this->promise);
```

Comment:
Do we ever need to put `this->promise` back? We don't always call `parent.drain` here.

Reply:
but we always call reject below.

Reply:
same for source, we always reject it.